mutable_buffer/src/table.rs

```diff
@@ -205,31 +201,38 @@ impl Table {
         }
 
         // make sure all the columns are of the same length
-        for col in &mut self.columns {
+        for col in self.columns.values_mut() {
             col.push_none_if_len_equal(row_count);
         }
 
         Ok(())
     }
 
     pub fn row_count(&self) -> usize {
-        self.columns.first().map_or(0, |v| v.len())
+        if let Some(col) = self.columns.values().next() {
+            return col.len();
+        }
+
+        0
     }
 
     /// The approximate memory size of the data in the table, in bytes. Note
     /// that the space taken for the tag string values is represented in the
     /// dictionary size in the chunk that holds the table.
     pub fn size(&self) -> usize {
-        self.columns.iter().fold(0, |acc, val| acc + val.size())
+        let mut size = 0;
+        for v in self.columns.values() {
+            size += v.size();
+        }
+        size
```

Comment:
again, not that it matters but why not something like

```suggestion
        self.columns.values().fold(0, |acc, val| acc + val.size())
```

Reply:
Also, much cleaner, will update.